examples/xoxo/xoxo.py

```diff
@@ -0,0 +1,183 @@
+
+def convert_num_to_indices(num):
+    row = int(num/sq_size)
+    col = num % sq_size
+    return(row,col)
+
+def already_played(row, col):
+    if display_board[row][col] == Cell.s:
+        return False
+    return True
+
+def play_move(player, row, col):
+    display_board[row][col] = player
+    solver.add_assertion(Equals(board[row][col], Int(player.value)))
+
+while True:
+
+    # get user input and handle errors
+    logger.info("-" * 40)
+    logger.info("turn %d" % turns)
+    print_board()
+    try:
+        next_cell = int(raw_input("type a cell (1-9):")) - 1
+    except ValueError:
+        continue
+    if next_cell < 0 or next_cell > 8:
+        continue
+
+    # convert index to rows, cols, check if space is free
+    row, col = convert_num_to_indices(next_cell)
+    if(not already_played(row, col)):
+        play_move(Cell.x, row, col)
+    else:
+        logger.info("that cell is already taken")
+        continue
+
+    turns += 1
+
+    # assertions for next move
+    x_next_move_assertion = Equals(Int((turns+1)*Cell.x.value + (turns*Cell.o.value)), Plus(Plus(board[0]), Plus(board[1]), Plus(board[2])))
+    x_this_move_assertion = Equals(Int((turns-1)*(Cell.x.value + Cell.o.value)+Cell.x.value), Plus(Plus(board[0]), Plus(board[1]), Plus(board[2])))
+    this_move_assertion = Equals(Int((turns)*(Cell.x.value + Cell.o.value)), Plus(Plus(board[0]), Plus(board[1]), Plus(board[2])))
+    o_win_this_turn_assertions = And(Or(o_win_assertions), this_move_assertion)
+    x_win_next_turn_assertions = And(Or(x_win_assertions), x_next_move_assertion)
+    x_win_this_turn_assertions = And(Or(x_win_assertions), x_this_move_assertion)
+
+    # detect x has won
+    logger.debug("checking if x has won")
+    logger.debug(x_win_this_turn_assertions)
+    if solver.solve([x_win_this_turn_assertions]):
+        print_solver_board()
+        exit("x wins")
+
+    # can I win next turn?
+    logger.debug("look for o to win")
+    logger.debug(o_win_this_turn_assertions)
+    if solver.solve([o_win_this_turn_assertions]):
+        logger.debug("o can win like this")
+        print_solver_board()
+        result = get_a_move(Cell.o)
+        if result is not None:
+            play_move(Cell.o, result[0], result[1])
+            print_board()
+            exit("o wins")
+
+    logger.debug("look for x to win next turn")
+    logger.debug(x_win_next_turn_assertions)
+    if solver.solve([x_win_next_turn_assertions]):
+        # can x win next turn? if so stop
+        logger.debug("x can win like this")
+        print_solver_board()
+        result = get_a_move(Cell.x)
+        if result is not None:
+            play_move(Cell.o, result[0], result[1])
+            continue
+
+    # otherwise try and win
+    logger.debug("looking for a win")
+    played = False
+    for future_turn in range(turns, 5):
+        move_assertion = Equals(Int((future_turn)*(Cell.x.value + Cell.o.value)), Plus(Plus(board[0]), Plus(board[1]), Plus(board[2])))
```

Comment:
Break this into a function to make the expression clear.

Reply:
same as above? Am I missing something? How would breaking this into a function make it clearer?